MerlinAU.sh

```diff
@@ -3139,17 +3139,18 @@ _EstimateNextCronTimeAfterDate_()
     while [ "$found" = "false" ]
     do
         loopCount="$((loopCount + 1))"
+
         if matches_month "$current_month" "$month_field" && \
            matches_day_of_month "$current_day" "$dom_field" && \
            matches_day_of_week "$current_dow" "$dow_field"
         then
             for this_hour in $(expand_cron_field "$hour_field" 0 23)
             do
-                if [ "$this_hour" -gt "$current_hour" ]
+                if [ "$this_hour" -gt "$current_hour" ] || [ "$loopCount" -gt 1 ]
                 then
                     for this_min in $(expand_cron_field "$minute_field" 0 59)
                     do
-                        echo "$(date '+%s' -d "$current_year-$current_month-$current_day $this_hour:$this_min")"
+                        echo "$(date -d "@$(date '+%s' -d "$current_year-$current_month-$current_day $this_hour:$this_min")" '+%Y-%m-%d %H:%M:%S')"
```

Comment:
It looks like you changed the format of the value being reported & eventually how it's stored in the configuration file: from a raw number of seconds to human-readable date/time stamp. I like it!!

Reply:
I did it for 2 reasons.

It seemed redundant to store in seconds and converted back. (It was less code to do it this way)

And second it follows the other date formats in the configuration file if you happen it open it up.

Reply:
>
> 
> And second it follows the other date formats in the configuration file if you happen it open it up.
>
Yeah, the "F/W Update Notification Date" is stored in human-readable form as well so they're consistent.